slsa/v1.0/L1/buildtype.md

```diff
@@ -0,0 +1,115 @@
+<!--
+SPDX-FileCopyrightText: 2022-2024 TII (SSRC) and the Ghaf contributors
+SPDX-License-Identifier: CC-BY-SA-4.0
+-->
+
+# Build Type: Jenkins
+
+This is a [SLSA Provenance](https://slsa.dev/provenance/v1) `buildType` that describes the execution of a Jenkins workflow.
+
+This build type was inspired by the Cimon Jenkins build type described [here](https://docs.cimon.build/provenance/buildtypes/jenkins/v1).
+
+## Description
+
+```json
+"buildType": "https://github.com/tiiuae/ghaf-infra/blob/main/slsa/v1.0/L1/buildtype.md"
+```
+
+The Jenkins build type describes the execution of a Jenkins workflow that builds a software artifact.
+
+## Build Definition
+
+### External parameters
+
+All external parameters are REQUIRED unless empty.
+
+Parameter             | Type   | Description
+--------------------- | ------ | -----------
+target                | object | The target that was built
+target.name           | string | The full name of the nix target
+target.repository     | string | URI of the git repository (if exists)
+target.ref            | string | A git reference to the commit (if exists)
+workflow              | object | The workflow that was run
+workflow.name         | string | The full name of the Jenkins job
+workflow.repository   | string | URI of the git repository (if exists)
+workflow.ref          | string | A git reference to the commit (if exists)
+workflow.filePath     | string | -
+job                   | object | The currently running Jenkins job
+buildRun              | object | The specific build that generated the provenance
+
+Example:
+
+```json
+"externalParameters": {
+    "target": {
+        "name": "#packages.x86_64-linux.generic-x86_64-debug",
+        "repository": "https://github.com/tiiuae/ghaf.git",
+        "ref": "main"
+    },
+    "workflow": {
+        "name": "ghaf-build-pipeline.groovy",
+        "repository": "https://github.com/tiiuae/ghaf-jenkins-pipeline.git",
+        "ref": "dc06ce39c110c1e997f15976842c8efa55b43825"
+    },
+    "job": "ghaf-pipeline",
+    "buildRun": "3"
+}
+```
+
+### Internal parameters
+
+All internal parameters are OPTIONAL. This build type doesn't use internal parameters.
+
+### Resolved Dependencies
+
+The resolvedDependencies array MUST contain all the dependencies of the target derivation, as reported by `nix-store --query --requisites`.
+
+The dependency's URI is a nix store path, which can be used to download the derivation from our binary cache at <https://cache.vedenemo.dev>.
```

Comment:
Is cache.vedenemo.dev used for azure ghaf-infra production builds? I were under impression that not (at the moment, at least)

Reply:
Currently the binary cache is available at https://ghaf-binary-cache-prod.northeurope.cloudapp.azure.com
It is _possible_ that down the road we'll change cache.vedenemo.dev point to the same cache instance but has not happened yet. For now it would be correct to use the above mentioned Azure URL.

Reply:
you're right I don't know why I used vedenemo cache here, got the systems mixed up